docs/galaxy_releases.rst

```diff
@@ -0,0 +1,219 @@
+    :code:`ansible-galaxy collection build`
+#.  By using the :code:`ansible-galaxy collection publish` command after the
+    build command.
+
+Ansible Collection Build Process
+--------------------------------
+
+
+Create the :code:`build/` directory.
+::
+
+  build
+  ├── galaxy_deploy.yml
+  └── templates
+      └── galaxy.yml.j2
+
+Instead of having the :code:`galaxy.yml` file in our root, we will be generating
+this each time we create a release.
+
+You will need to create the :code:`galaxy_deploy.yml` Ansible playbook which
+will handle the build and publish steps for us. It's contents should look
+something like this.
+
+galaxy_deploy.yml
+^^^^^^^^^^^^^^^^^
+.. code-block:: yaml
+
+  ---
+  - hosts: localhost
+    connection: local
+    gather_facts: false
+    vars:
+      # This should be set via the command line at runtime.
+      tag: "{{ github_tag.split('/')[-1] }}"
+    pre_tasks:
+      - name: Ensure the ANSIBLE_GALAXY_TOKEN environment variable is set.
+        fail:
+          msg: ANSIBLE_GALAXY_TOKEN is not set.
+        when: "lookup('env','ANSIBLE_GALAXY_TOKEN') | length == 0"
+      - name: Ensure the ~/.ansible directory exists.
+        file:
+          path: ~/.ansible
+          state: directory
+      - name: Write the Galaxy token to ~/.ansible/galaxy_token
+        copy:
+          content: |
+            token: {{ lookup('env','ANSIBLE_GALAXY_TOKEN') }}
+          dest: ~/.ansible/galaxy_token
+          mode: '0600'
+    tasks:
+      - name: Template out the galaxy.yml file.
+        template:
+          src: templates/galaxy.yml.j2
+          dest: ../galaxy.yml
+        register: galaxy_yml
+      - name: Build the collection. # noqa 503
+        command: >
+          ansible-galaxy collection build
+          chdir=../
+        when: galaxy_yml.changed
+      - name: Publish the collection. # noqa 503
+        command: >
+          ansible-galaxy collection publish ./namespace-collection-{{ tag }}.tar.gz
+          chdir=../
+        when: galaxy_yml.changed
+
+You'll then need to create the :code:`build/templates` directory which we will
+be placing our :code:`galaxy.yml.j2` jinja template. Please edit the jinja
+template to your liking. Do not change the :code:`version` section as this is
+necessary for Ansible to properly template your :code:`galaxy.yml` for the
+specific release.
+
+galaxy.yml.j2
+^^^^^^^^^^^^^
+.. code-block:: yaml
+
+  ### REQUIRED
+
+  # The namespace of the collection. This can be a company/brand/organization or product namespace under which all
+  # content lives. May only contain alphanumeric lowercase characters and underscores. Namespaces cannot start with
+  # underscores or numbers and cannot contain consecutive underscores
+  namespace: namespace_name
+
+  # The name of the collection. Has the same character restrictions as 'namespace'
+  name: collection_name
+
+  # The version of the collection. Must be compatible with semantic versioning
+  version: "{{ tag }}"
```

Comment:
AFAICS this is the only value that's being rendered in this template, it seems to be overengineered and could be solved with a simple `sed`.

Reply:
Awww, it's like we don't even need Ansible for things ;)

Reply:
But if you notice tag isn't provided by the script, it's actually being processed as github.ref includes `/` inside because it's a release ref, not a number.

Reply:
yep, I saw that. Btw you could easily get it from the Git checkout like `setuptools-scm` does this which is even better because this'd be portable and could run from the local machine as well

Reply:
I was just using the params that github provides internally, are you able to use github actions locally? or should we just delete this PR and move to using tags and using more than github workflow for releases?

Reply:
I said above assuming you're adding this extra ansible playbook layer because you want this automation to be portable and run both locally (w/o  GH actions) and in CI/CD.
If you only intend to have a GHA based solution you don't really need this many levels of indirection.